src/acvp_kdf135_ikev2.c

```diff
@@ -35,12 +35,56 @@
  * Forward prototypes for local functions
  */
 static ACVP_RESULT acvp_kdf135_ikev2_output_tc (ACVP_CTX *ctx, ACVP_KDF135_IKEV2_TC *stc, JSON_Object *tc_rsp) {
-    json_object_set_string(tc_rsp, "sKeySeed", (const char *)stc->s_key_seed);
-    json_object_set_string(tc_rsp, "sKeySeedReKey", (const char *)stc->s_key_seed_rekey);
-    json_object_set_string(tc_rsp, "derivedKeyingMaterial", (const char *)stc->derived_keying_material);
-    json_object_set_string(tc_rsp, "derivedKeyingMaterialChild", (const char *)stc->derived_keying_material_child);
-    json_object_set_string(tc_rsp, "derivedKeyingMaterialDh", (const char *)stc->derived_keying_material_child_dh);
-    return ACVP_SUCCESS;
+    ACVP_RESULT rv = ACVP_SUCCESS;
+    char *tmp;
+    
+    tmp = calloc(ACVP_KDF135_IKEV2_SKEY_SEED_STR_MAX, sizeof(char));
+    rv = acvp_bin_to_hexstr(stc->s_key_seed, stc->key_out_len, (unsigned char *) tmp);
+    if (rv != ACVP_SUCCESS) {
+        ACVP_LOG_ERR("hex conversion failure (s_key_seed)");
+        goto err;
+    }
+    json_object_set_string(tc_rsp, "sKeySeed", (const char *)tmp);
+    memset(tmp, 0x0, ACVP_KDF135_IKEV2_SKEY_SEED_STR_MAX);
+    
+    rv = acvp_bin_to_hexstr(stc->s_key_seed_rekey, stc->key_out_len, (unsigned char *) tmp);
```

Comment:
You can use the new #defines here for the length check! https://github.com/cisco/libacvp/commit/c092c1d48c24f15efc3cac6eb5fd6fd7ec0ae0f1 Make sure to use from current code becuas that commit has a little bug (that was fixed in a later commit)

Reply:
Most of these are the new ones, but there were a handful that I didn't catch. Thanks!!